src/microservices/Microservices.IsIdentifiable/Runners/IsIdentifiableAbstractRunner.cs

```diff
@@ -88,6 +89,26 @@ public abstract class IsIdentifiableAbstractRunner : IDisposable
         /// </summary>
         public List<ICustomRule> CustomWhiteListRules { get; set; } = new List<ICustomRule>();
 
+        /// <summary>
+        /// One cache per field in the data being evaluated, records the recent values passed to <see cref="Validate(string, string)"/> and the results to avoid repeated lookups
+        /// </summary>
+        public ConcurrentDictionary<string,ConcurrentDictionary<string,FailurePart[]>> Caches {get;set;} = new ConcurrentDictionary<string, ConcurrentDictionary<string, FailurePart[]>>();
+
+        /// <summary>
+        /// The maximum size of a Cache before we clear it out to prevent running out of RAM
+        /// </summary>
+        private const int MaxCacheSize = 1_000_000;
```

Comment:
Are we ok with this being a constant, or should we make it configurable somehow?

Reply:
Now an option `MaxValidationCacheSize` on CLI:
```
 --maxvalidationcachesize         (Default: 1000000) Optional.  Maximum number
                                   of answers to cache per column.
```